### ISSUE TITLE
no-unused-vars not working as it should

### ISSUE THREAD

**Issue by LewisMcMahon** [eslint/eslint#5727]
**What version of ESLint are you using?**
v2.5.3

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```js
module.exports = {
    "env": {
        "node": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            "error",
            "tab"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
        ],
        "no-console":0

    }
};
```

**What did you do? Please include the actual source code causing the issue.**
 'deltaTime' is defined but never used no-unused-vars yet it is used in the else statement

```js
function clientTickFunction() {
		//calculate delta time and set it for next time
		var deltaTime = 0;
		var currentTimestamp = Date.now();
		if (!lastClientTick) {
			lastClientTick = currentTimestamp;
		}
		else {
			deltaTime = (currentTimestamp - lastClientTick) / 1000;
		}
		lastClientTick = currentTimestamp;

		//loop through all clients and update them on the game state
		clients.forEach(function (client) {
			client.emit("ShipData",Orion.consoleObjects["Engineering"].getData());
		}, this);

		return true;
	}
```

**What did you expect to happen?**
nothing this code should pass

**What actually happened? Please include the actual, raw output from ESLint.**
note the above code is taken from a large block so line numbers are not correct but line 82 corresponds to line 3 above
```sh
  82:7  error  'deltaTime' is defined but never used  no-unused-vars

? 1 problem (1 error, 0 warnings)

```

**Comment by platinumazure** [comment]
The `deltaTime` variable isn't used, only assigned to. Furthermore, it's declared in the same scope and I don't see any closure which uses it. I think the lint error is correct and this is a sign that you might need to ensure you are using `deltaTime`.

**Comment by ilyavolodin** [comment]
@LewisMcMahon variable is considered as used when it's assigned to another variable or used in a function call. You are just assigning a value to that variable, but that value is not used anywhere in the code you provided. The rule is doing the right thing in this case.

**Comment by LewisMcMahon** [comment]
Turns out im an idiot ill close this
